ibis/backends/tests/test_client.py

```diff
@@ -202,3 +203,263 @@ def test_separate_database(con, alternate_current_database, current_data_db):
     db = con.database(current_data_db)
     assert db.name == current_data_db
     assert tmp_db.name == alternate_current_database
+
+
+def create_temp_table_with_schema(con, temp_table_name, schema):
+
+    con.drop_table(temp_table_name, force=True)
+    con.create_table(temp_table_name, schema=schema)
+    temporary = con.table(temp_table_name)
+    assert len(temporary.execute()) == 0
+
+    return temporary
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_append_from_dataframe(con):
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    temp_table = 'temp_to_table'
+    temporary = create_temp_table_with_schema(con, temp_table, sch)
+
+    records = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
+
+    con.insert(temp_table, data_obj=records, if_exists='append')
+    assert len(temporary.execute()) == 3
+    tm.assert_frame_equal(temporary.execute(), records)
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_replace_from_dataframe(con):
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    temp_table = 'temp_to_table'
+    temporary = create_temp_table_with_schema(con, temp_table, sch)
+
+    df = pd.DataFrame(
+        {
+            'first_name': ['A', 'B', 'C'],
+            'last_name': ['D', 'E', 'F'],
+            'department_name': ['AA', 'BB', 'CC'],
+            'salary': [100.0, 200.0, 300.0],
```

Comment:
Would it makes sense to use always the same data, and have this, the loading on the data, and the assert that the right number of rows are there in the create table function you added?

I guess there is a reason I'm missing to have very similar data but different. But would make things much more compact and readable if we could have all the setup in the function, and in the tests just the specific things we want to test in each case.

Reply:
In some cases we want the `to_table` to be empty so that we can not only assert the number of rows in the final table but also the columns and other info using `assert_frame_equal`. `assert_frame_equal` can't be used when there are extra rows present compared to the `from_table`.  
I'll try to improvise the `_create_temp_table_with_schema` function to take in data as well but can't make the data part aslo go into this function since it needs to be reused.